src/rest-server/src/models/v2/virtual-cluster/k8s.js

```diff
@@ -32,6 +32,26 @@ const vcData = {
 };
 
 
+const fetchNodes = async () => {
+  const nodes = await axios({
+    method: 'get',
+    url: `${kubernetesConfig.apiserver.uri}/api/v1/nodes`,
+  });
+  return nodes.data.items.filter((node) => {
+    if (node.metadata.labels['pai-worker'] !== 'true') {
```

Comment:
filter with "pai-worker" might be unreliable. what if the admin removes the label? right now FC can still use a node without pai-worker.

Reply:
`pai-worker` label is used to remove master nodes and machines not maintained by pai,
since the quota is shown on pai's webportal

for readiness check, the following field is used:
```jsonp
{
  "type": "Ready",
  "status": "True",
  "lastHeartbeatTime": "2019-09-30T07:26:19Z",
  "lastTransitionTime": "2019-09-30T07:12:51Z",
  "reason": "KubeletReady",
  "message": "kubelet is posting ready status"
}
```
when the node is preempted or kubelet is killed, it will become
```jsonp
{
  "type": "Ready",
  "status": "Unknown",
  ...
}
```

Reply:
sounds like we need to add a node selector when submitting jobs from pai?

currently, jobs may be scheduled to a master node, which is not by design
it is scheduled by default scheduler or hived scheduler, instead of framework controller

Reply:
In this PR, maybe we can just fix the "unknown" node bug. i.e., when you calculate the available nodes, please excludes those unhealthy nodes.

Reply:
yes, the pai-worker label filter is added at the beginning, just added a check for not ready nodes and moved it to a function